apps/wallet/src/ui/app/shared/collapse/index.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { AnimatePresence, motion } from 'framer-motion';
+import { useState } from 'react';
+
+import Icon, { SuiIcons } from '_components/icon';
+
+import type { ReactNode } from 'react';
+
+type CollapseProps = {
+    title: string;
+    initialIsOpen?: boolean;
+    children: ReactNode | ReactNode[];
+};
+
+export function Collapse({
```

Comment:
Makes sense, I'm planning on adding real base UI components support here so it might make sense to start doing this and extract this kind of stuff, but I'll wait to review it that way until we really have it all setup.